mne/chpi.py

```diff
@@ -149,6 +150,115 @@ def head_pos_to_trans_rot_t(quats):
     return translation, rotation, t
 
 
+def _apply_quat(quat, pts, move=True):
+    """Apply MaxFilter-formatted head position parameters to points."""
+    trans = np.concatenate(
+        (quat_to_rot(quat[:3]),
+         quat[3:][:, np.newaxis]), axis=1)
+
+    return(apply_trans(trans, pts, move=move))
+
+
+def extract_head_pos_ctf(raw, gof_limit=0.98):
+    """Extract MaxFilter-formatted head position parameters from ctf dataset.
+
+    Parameters
+    ----------
+    raw : instance of raw
+        Raw data with cHPI information. HLC00 channels
+    gof_limit : float
+        Minimum goodness of fit to accept.
+
+    Returns
+    -------
+    pos : array, shape (N, 10)
+        The position and quaternion parameters from cHPI fitting.
+
+    See Also
+    --------
+
+    Notes
+    -----
+    CTF continuous head monitoring stores the x,y,z location (m) of each chpi
+    coil as separate channels in the dataset.
+    HLC001[123]-* - nasion
+    HLC002[123]-* - lpa
+    HLC003[123]-* - rpa
+    """
+    # make picks to match order of dig cardinial ident codes.
+    # LPA, NAS, RPA
+    hpi_picks = []
+    for i in [2, 1, 3]:
+        hpi_picks.extend(pick_channels_regexp(raw.info['ch_names'],
+                                              'HLC00%d[123]-.*' % i))
```

Comment:
I don't see any checking that `pick_channels_regexp` actually gives you channels in the right order.

You might just want to pick channels with a single `regex`, then check that you actually have the correct channels while doing the necessary reordering.